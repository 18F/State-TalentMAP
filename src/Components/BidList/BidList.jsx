import React from 'react';
import PropTypes from 'prop-types';
import * as PROP_TYPES from '../../Constants/PropTypes';
import Spinner from '../Spinner';
import BidListResultsList from '../BidListResultsList';
import Alert from '../Alert/Alert';
import ProfileSectionTitle from '../ProfileSectionTitle';

const SavedSearches = ({ bidList, toggleBidPosition,
bidListHasErrored, bidListIsLoading, bidListToggleHasErrored,
bidListToggleIsLoading, bidListToggleSuccess }) => {
  const showError = !bidListToggleIsLoading && !bidListToggleSuccess
      && bidListToggleHasErrored;
  const showSuccess = !bidListToggleIsLoading && bidListToggleSuccess
      && !bidListToggleHasErrored;
  const isLoading = bidListIsLoading && !bidListHasErrored;
  return (
    <div
      className={`usa-grid-full saved-searches-container
    ${bidListIsLoading ? 'results-loading' : ''}`}
    >
      <ProfileSectionTitle title="Your Bid List:" />
      {
        // Deleting a bid has errored
        showError &&
          <Alert
            type="error"
            title="Error"
            messages={[{ body: bidListToggleHasErrored }]}
          />
      }
      {
        // Deleting a bid was successful
        showSuccess &&
          <Alert
            type="success"
            title="Success"
            messages={[{ body: bidListToggleSuccess }]}
          />
      }
      {
        isLoading &&
        <Spinner type="homepage-position-results" size="big" />
      }
      <BidListResultsList
        bidList={bidList}
        toggleBidPosition={toggleBidPosition}
      />
    </div>
  );
};

SavedSearches.propTypes = {
  toggleBidPosition: PropTypes.func.isRequired,
  bidListHasErrored: PropTypes.bool.isRequired,
  bidListIsLoading: PropTypes.bool.isRequired,
  bidList: PROP_TYPES.BID_LIST.isRequired,
  bidListToggleHasErrored: PropTypes.bool.isRequired,
  bidListToggleIsLoading: PropTypes.bool.isRequired,
  bidListToggleSuccess: PropTypes.bool.isRequired,
};

export default SavedSearches;
