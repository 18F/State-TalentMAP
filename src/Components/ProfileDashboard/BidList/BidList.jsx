import React from 'react';

const BidList = () => (
  <div className="usa-grid-full">
    Bid list
  </div>
);

export default BidList;
