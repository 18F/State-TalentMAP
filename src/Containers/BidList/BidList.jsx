import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import * as bidListActions from '../../actions/bidList';
import * as PROP_TYPES from '../../Constants/PropTypes';
import BidList from '../../Components/BidList';

class BidListContainer extends Component {

  componentWillMount() {
    this.getBidList();
    // reset the alert messages
    this.props.bidListRouteChangeResetState();
  }

  getBidList() {
    this.props.fetchBidList();
  }

  render() {
    const { bidList, toggleBidPosition,
      bidListHasErrored, bidListIsLoading, bidListToggleHasErrored,
      bidListToggleIsLoading, bidListToggleSuccess, submitBid,
      submitBidHasErrored, submitBidIsLoading, submitBidSuccess } = this.props;
    return (
      <div>
        <BidList
          toggleBidPosition={toggleBidPosition}
          bidList={bidList}
          bidListHasErrored={bidListHasErrored}
          bidListIsLoading={bidListIsLoading}
          bidListToggleHasErrored={bidListToggleHasErrored}
          bidListToggleIsLoading={bidListToggleIsLoading}
          bidListToggleSuccess={bidListToggleSuccess}
          submitBid={submitBid}
          submitBidHasErrored={submitBidHasErrored}
          submitBidIsLoading={submitBidIsLoading}
          submitBidSuccess={submitBidSuccess}
        />
      </div>
    );
  }
}

BidListContainer.propTypes = {
  bidListRouteChangeResetState: PropTypes.func.isRequired,
  fetchBidList: PropTypes.func,
  toggleBidPosition: PropTypes.func,
  bidListHasErrored: PropTypes.bool,
  bidListIsLoading: PropTypes.bool,
  bidList: PROP_TYPES.BID_LIST,
  bidListToggleHasErrored: PROP_TYPES.BID_LIST_TOGGLE_HAS_ERRORED,
  bidListToggleIsLoading: PropTypes.bool,
  bidListToggleSuccess: PROP_TYPES.BID_LIST_TOGGLE_SUCCESS,
  submitBid: PropTypes.func.isRequired,
  submitBidHasErrored: PROP_TYPES.SUBMIT_BID_HAS_ERRORED.isRequired,
  submitBidIsLoading: PropTypes.bool.isRequired,
  submitBidSuccess: PROP_TYPES.SUBMIT_BID_SUCCESS.isRequired,
};

BidListContainer.defaultProps = {
  fetchBidList: PROP_TYPES.EMPTY_FUNCTION,
  toggleBidPosition: PROP_TYPES.EMPTY_FUNCTION,
  bidList: { results: [] },
  bidListHasErrored: false,
  bidListIsLoading: false,
  bidListToggleHasErrored: false,
  bidListToggleIsLoading: false,
  bidListToggleSuccess: false,
  submitBid: PROP_TYPES.EMPTY_FUNCTION,
  submitBidHasErrored: false,
  submitBidIsLoading: false,
  submitBidSuccess: false,
};

BidListContainer.contextTypes = {
  router: PropTypes.object,
};

const mapStateToProps = state => ({
  bidListHasErrored: state.bidListHasErrored,
  bidListIsLoading: state.bidListIsLoading,
  bidList: state.bidListFetchDataSuccess,
  bidListToggleHasErrored: state.bidListToggleHasErrored,
  bidListToggleIsLoading: state.bidListToggleIsLoading,
  bidListToggleSuccess: state.bidListToggleSuccess,
  submitBidHasErrored: state.submitBidHasErrored,
  submitBidIsLoading: state.submitBidIsLoading,
  submitBidSuccess: state.submitBidSuccess,
});

const mapDispatchToProps = dispatch => ({
  fetchBidList: () => dispatch(bidListActions.bidListFetchData()),
  toggleBidPosition: (id, remove) => dispatch(bidListActions.toggleBidPosition(id, remove)),
  bidListRouteChangeResetState: () => dispatch(bidListActions.routeChangeResetState()),
  submitBid: id => dispatch(bidListActions.submitBid(id)),
});

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(BidListContainer));
