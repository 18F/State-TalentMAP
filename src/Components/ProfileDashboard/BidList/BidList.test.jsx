import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import BidList from './BidList';
import bidListObject from '../../../__mocks__/bidListObject';

describe('BidListComponent', () => {
  it('is defined', () => {
    const wrapper = shallow(<BidList bids={bidListObject.results} />);
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<BidList bids={bidListObject.results} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
