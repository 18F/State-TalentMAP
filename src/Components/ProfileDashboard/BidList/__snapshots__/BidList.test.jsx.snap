// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BidListComponent matches snapshot when editor is shown 1`] = `
<div
  className="usa-grid-full"
>
  Bid list
</div>
`;
